qa/rpc-tests/wallet.py

```diff
@@ -89,6 +89,26 @@ def run_test (self):
         assert_equal(len(node2utxos), 2)
         assert_equal(sum(int(uxto["generated"] is True) for uxto in node2utxos), 0)
 
+        # Catch an attempt to send a transaction with an absurdly high fee.
+        # Send 1.0 from an utxo of value 10.0 but don't specify a change output, so then
+        # the change of 9.0 becomes the fee, which is greater than estimated fee of 0.0019.
+        inputs = []
+        outputs = {}
+        for utxo in node2utxos:
+            if utxo["amount"] == Decimal("10.0"):
+                break
+        assert_equal(utxo["amount"], Decimal("10.0"))
```

Comment:
It somewhat irks me that Python leaves the last iteration of a loop in scope. Handy, yes, but...

Reply:
Yes! I think this is related to another annoyance of Python:
```
for i in range(3):
    print(i)
print(i)
```
produces this output:
```
0
1
2
2
```
It seems like the final value of `i` should be 3 (as in C), not 2. If it were 3, then you'd know you didn't break out of the loop early (not being done in this example, but in general).